matrix-sdk/src/main/java/org/matrix/androidsdk/data/Room.java

```diff
@@ -2102,7 +2102,107 @@ public void sendEvent(final Event event, final ApiCallback<Void> callback) {
 
         final String prevEventId = event.eventId;
 
-        final ApiCallback<CreatedEvent> localCB = new ApiCallback<CreatedEvent>() {
+        final ApiCallback<CreatedEvent> localCB = getLocalCB(event, callback, prevEventId);
+
+        if (isEncrypted() && (null != mDataHandler.getCrypto())) {
```

Comment:
Seems like a long method. Some of the if-else's could potentially be extracted into their own methods.